snuba/query/processors/logical/filter_in_select_optimizer.py

```diff
@@ -1,234 +1,145 @@
+                    "Failed during optimization", exc_info=True, extra={"query": query}
                 )
-                domain = {}
-
-            if domain:
-                # add domain to where clause
-                domain_filter = None
-                for key, value in domain.items():
-                    clause = binary_condition(
-                        "in",
-                        key,
-                        FunctionCall(
-                            alias=None,
-                            function_name="array",
-                            parameters=tuple(value),
-                        ),
-                    )
-                    if not domain_filter:
-                        domain_filter = clause
-                    else:
-                        domain_filter = binary_condition(
-                            "and",
-                            domain_filter,
-                            clause,
-                        )
-                assert domain_filter is not None
-                query.add_condition_to_ast(domain_filter)
-                metrics.increment("kyles_optimizer_optimized")
-
-    def get_domain_of_mql_query(
-        self, query: LogicalQuery | CompositeQuery[QueryEntity]
-    ) -> Domain:
+                return
+
+    def get_select_filter(
+        self,
+        query: LogicalQuery | CompositeQuery[QueryEntity],
+    ) -> FunctionCall | None:
         """
-        This function returns the metric_id domain of the given query.
-        For a definition of metric_id domain, go to definition of the return type of this function ('Domain')
+        Given a query, grabs all the conditions from conditional aggregates and lifts into
+        one condition.
+
+        ex: SELECT sumIf(value, metric_id in (1,2,3,4) and status=200),
+                   avgIf(value, metric_id in (11,12) and status=400),
+                   ...
+
+        returns or((metric_id in (1,2,3,4) and status=200), (metric_id in (11,12) and status=400))
         """
-        expressions = map(lambda x: x.expression, query.get_selected_columns())
-        target_exp = None
-        for exp in expressions:
-            if self._contains_conditional_aggregate(exp):
-                if target_exp is not None:
+        # find and grab all the conditional aggregate functions
+        cond_agg_functions: list[FunctionCall | CurriedFunctionCall] = []
+        for selected_exp in query.get_selected_columns():
+            exp = selected_exp.expression
+            finder = FindConditionalAggregateFunctionsVisitor()
+            exp.accept(finder)
+            found = finder.get_matches()
+            if len(found) > 0:
+                if len(cond_agg_functions) > 0:
                     raise ValueError(
-                        "Was expecting only 1 select expression to contain condition aggregate but found multiple"
+                        "expected only one selected column to contain conditional aggregate functions but found multiple"
```

Comment:
This edge case is not clear to me why it exists, the error message is not helpful for the user. It's also not a tested case. Also if this is actually an invalid query, you should raise `snuba.query.exceptions.InvalidQueryException`

Reply:
Reading this more, I don't think this is invalid. If there are multiple `If` conditions, their metric ids should just be merged together

Reply:
removed